{Conesa:2016du} have a nice RNA-seq review, that includes various QC issues, as well as a discussion of advanced methods such as single-cell analyses, small RNAs and fusion transcripts

{Ameur:2011jy} find lots of unprocessed mRNA in fetal brain samples, and show transcriptional splicing, where introns are removed during transcription. The signature for this is a saw-tooth pattern where the 5' ends of large introns have a lot of read data, but the 3' ends don't because the introns are removed as soon as the 3' end is transcribed. This was also confirmed with PCR across splice sites. I think this mostly indicates a high level of mRNA turn-over and high levels of active transcription. 

- De novo splice junctions were detected using version 1.3.2 of SplitSeek
    - this tool appears to be a bit rubbish

"To identify transcripts novel as compared with Ensembl, we utilized Cuffcompare class codes and extracted those assembled transcripts classified as: i – novel intronic, u – novel intergenic, x – novel anti- sense. All novel transcripts under 200 nt in length were removed. Of the remaining transcripts, we determined minimal read coverage thresholds based on whether Cufflinks classified previously annotated transcripts as having “full_read_support.” By analyzing the true positive rate versus false positive rate of classifying known genes as obtaining “full_read_support” at various coverage thresh- olds, we determined the minimum coverage to be 1.4 for polyA and 1.67 for total RNA-seq (at FDR = 0.05). Starting with just the polyA RNA-seq data, transcripts with read coverage above 1.4 in both biological replicates of at least one developmental stage were included in the reference and considered to be expressed in the neocor- tex." {Liu:2016ji}
