---
title: "Data QC"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(RColorBrewer)
library(tufte)
rm(list=ls())
source("~/BTSync/Code/R/FormatGGplot.R")
folders <- c("/Users/heo3/BTSync/FetalRNAseq/BamQC/15240/15240",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/15240.wasp",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/15533/15533",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/15655/15655",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/15655.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16286/16286",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16286.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16428/16428",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16428.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16438/16438",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16438.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16483/16483",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16483.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16640/16640",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16640.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16649/16649",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16649.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16810/16810",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16810.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16972/16972",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/16972.wasp",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17048/17048",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/Wasp/17048.wasp"
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17087/17087",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17109/17109",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17701/17701",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18294/18294",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18655/18655",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/15768/15768",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17054/17054",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17068/17068",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17081/17081",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17115/17115",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17130/17130",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17812/17812",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18349/18349",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18596/18596",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/15468/15468",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16488/16488",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16840/16840",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16929/16929",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17049/17049",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18687/18687",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/15641/15641",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16024/16024",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16115/16115",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16385/16385",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16491/16491",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16548/16548",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16810/16810",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/16826/16826",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17053/17053",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17071/17071",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17921-l1/17921-l1",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17013/17013",
             # < 2000 reads "/Users/heo3/BTSync/FetalRNAseq/BamQC/17025/17025",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17198/17198",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17229/17229",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17333/17333",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17475/17475",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17543/17543",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17629/17629",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17835/17835",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18249/18249", 
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18372/18372",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18666/18666",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18983/18983",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/19043/19043",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/19052/19052",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/A138/A138",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/A226/A226",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17072/17072",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17160/17160",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17167/17167",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17175/17175",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17671/17671",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17922/17922",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17923/17923",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18121/18121",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18134/18134",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18153/18153",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18241/18241",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18266/18266",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18355/18355",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18432/18432",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18559/18559",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18694/18694",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17369/17369",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18055/18055",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/15533_2/15533_2",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/18282/18282"
             )
```

# RSeQC
## Mapping Stats
`r margin_note("- between 32% and 47% lost when all SNPs in VCF used fort WASP remapping")`

``` {r warnings=FALSE }
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- suppressWarnings(tryCatch({
    read.delim(paste(folder, ".ex.stats.txt", sep=""), 
                     header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                     )
  }, error=function(er){
    read.delim(paste(folder, ".stats.txt", sep=""), 
                     header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                     )
  }))
  temp[4,2] <-strsplit(temp[4,1], ' +')[[1]][4]
  temp[4,1] <- 'Non primary hits'
  temp <- temp[c(6,7,14),]
  temp[,1] <- c("Multimapped", "Unique", "Paired")
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)
#mutate(RSeQCstats, Perc_rDNA = as.numeric(rDNA)/(as.numeric(Unique)+as.numeric(Multimapped)+as.numeric(rDNA))) %>% arrange(Perc_rDNA)
ReadNumbers <- RSeQCstats[,c(1,4)]
RSeQCstats <- RSeQCstats[c(1,4,3,2)]
RSeQCstats[,2:4] <- sapply(RSeQCstats[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))
knitr::kable(
  RSeQCstats
)
```


## Distribution of Reads Among Features (proportion of tags mapping to different feature classes)
`r margin_note("- In most cases, higher proportion of tags map to CDS and lower proportion map to introns after WASP remapping")`


``` {r }
RSeQCdistribution <- data.frame()
ReadNumbers$ExpressedTags = NA
for (folder in folders) {
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp <- read.delim(paste(folder, ".dist.txt", sep=""), 
                           header=TRUE, stringsAsFactors=FALSE, skip=4, sep=''
                     )
  temp <- temp[-11,]
  tag_total <- sum(temp[c(1,2,3,4,7,10),3])
  ReadNumbers[ReadNumbers$sample == sample,]$ExpressedTags <- sum(temp[c(1,2,3),3])
  temp <- data.frame(V1 = c("Tags_per_read", "CDS", "UTR", "Intron", "Intergenic"), 
                     V2=c(
                  tag_total/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique),
                  temp[1,3]/tag_total, 
                  sum(as.numeric(temp[c(2,3),3]))/tag_total,
                  temp[4,3]/tag_total,
                  sum(as.numeric(temp[c(7,10),3]))/tag_total
                  )
             )
  temp$sample <- sample
  #temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCdistribution <- rbind(RSeQCdistribution, temp)
}
RSeQCdistribution <- spread(RSeQCdistribution, V1, V2)
RSeQCdistribution <- RSeQCdistribution[c(1,5,2,6,4,3)]

knitr::kable(
  RSeQCdistribution
)
```

## Strand

`r margin_note("- proportion of first -strand and ambiguous reads goes down after WASP remapping")`

```{r }
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".expt.txt", sep=""), 
                     skip=3, header=FALSE, sep=':'
                     )
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$V1 <- c("Ambiguous", "First Strand", "Second Strand")
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)

knitr::kable(
  RSeQCstats
)
```

## Insert Size Distribution

`r margin_note("- These look like they match up exactly")`


```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  #print(folder)
  temp <- read.delim(paste(folder, ".inner_distance_freq.txt", sep=""),
                     header=FALSE
                     )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$mapping <- ifelse(grepl("wasp",folder), 'Wasp', 'All')
  RSeQCstats <- rbind(RSeQCstats, temp)
}
```

```{r dev.args = list(bg = 'transparent')}
ggplot(RSeQCstats, aes(x=(V1+V2)/2, y=V3, colour=mapping, group=sample)) +
  geom_line() +
  ylab('reads') +
  xlab('inner distance') +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  tufte_theme() 

```

## Transcript Read Coverage

`r margin_note("- For some reason, WASP remapping appears to clean up samples with low proportions. Dono")`

```{r dev.args = list(bg = 'transparent')}
RSeQCstats <- data.frame()
for (folder in folders) {
  #print(folder)
  temp <- read.delim(paste(folder, ".geneBodyCoverage.txt", sep=""),
                     header=FALSE
                     )
  temp <-  as.data.frame(t(temp[,-1]))
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$mapping <- ifelse(grepl("wasp",folder), 'Wasp', 'All')
  temp$V2 <- temp$V2
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$ExpressedTags)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

#ggplot(subset(RSeQCstats, sample != 17175), aes(x=V1, y=V2/ReadNumbers, colour=centre, group=sample)) +
ggplot(RSeQCstats, aes(x=V1, y=V2/ReadNumbers, colour=mapping, group=sample)) +
  geom_line() +
  ylab('reads') +
  tufte_theme() +
  theme(legend.position=c(0.1,.9)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  xlab('transcript percentile') +
  ylab("proportion of expressed tags")
  
```

## Sequence Duplication

`r margin_note("- Looks like WASP remapping reduces duplication slightely")`

```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".DupRate_plot.r", sep=""))[2:6])
  temp <- rbind(data.frame(Reads=pos_uniqRead, 
                            Duplication_level=pos_occ, 
                            Method='Position-based'
                            ),
                 data.frame(Reads=seq_uniqRead, 
                            Duplication_level=seq_occ, 
                            Method='Sequence-based'
                            )
                )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$mapping <- ifelse(grepl("wasp",folder), 'Wasp', 'All')
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(RSeQCstats, aes(x=Duplication_level, y=Reads/ReadNumbers, colour=mapping, group=sample)) +
  geom_point(alpha=.2) +
  xlim(c(0,500)) +
  scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  facet_wrap(~ Method) +
  ylab("Proportion of Reads")


```

## Splice Junction Saturation

`r margin_note("- WASP remapping drastically reduces the number of slice sites in some samples")`

```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".junctionSaturation_plot.r", sep=""))[2:5])
  temp<-rbind(
              data.frame(percent_reads=x, junctions=z, Category='All'),
              data.frame(percent_reads=x, junctions=y, Category='Known'),
              data.frame(percent_reads=x, junctions=w, Category='Novel')
              )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,2])
  temp$mapping <- ifelse(grepl("wasp",folder), 'Wasp', 'All')
  RSeQCstats <- rbind(RSeQCstats, temp)
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(RSeQCstats, aes(y=junctions, x=ReadNumbers*percent_reads/100, group=sample, colour=mapping)) +
  geom_point() +
  #xlim(c(0,500)) +
  #scale_y_log10() +
  scale_colour_brewer(type = "qual", palette = 6) +
  tufte_theme() +
  theme(legend.position=c(.8,.9)) +
  facet_wrap(~ Category, scales='free') +
  xlab("Number of Reads")

```

