---
title: "Data QC"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(RColorBrewer)
library(tufte)
rm(list=ls())
source("~/BTSync/Code/R/FormatGGplot.R")
```

# RSeQC
## Mapping Stats
`r margin_note("- Excludes reads mapping to rDNA")`
`r margin_note("- Total records = mapped reads + non-primary")`
`r margin_note("- VERY low number of proper pairs. This doesn't seem to be caused by the strand because it is the same for other strand options. I'm trying a different mean mapping dist to see if that helps")`

``` {r }
RSeQCstats <- read.delim("/Users/heo3/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.ex.stats.txt", header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats[4,2] <-strsplit(RSeQCstats[4,1], ' +')[[1]][4]
RSeQCstats[4,1] <- 'Non primary hits'
colnames(RSeQCstats)<-c('stat', 'result')
RSeQCstats$result <- formatC(as.numeric(RSeQCstats$result), format='d', big.mark=',')
knitr::kable(
  RSeQCstats
)
```

## rRNA Mapping Stats

`r margin_note("- 14 million reads map to rDNA (ca. 5%)")`
`r margin_note("- These mappings account for 20% of multimapped reads and >1/3 of all non-primary hits")`

``` {r }
RSeQCstats <- read.delim("/Users/heo3/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.in.stats.txt", header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats[4,2] <-strsplit(RSeQCstats[4,1], ' +')[[1]][4]
RSeQCstats[4,1] <- 'Non primary hits'
colnames(RSeQCstats)<-c('stat', 'result')
RSeQCstats$result <- formatC(as.numeric(RSeQCstats$result), format='d', big.mark=',')
knitr::kable(
  RSeQCstats
)
```


## Distribution of Reads Among Features
`r margin_note("- Exons enriched >10 fold relative to introns")`
`r margin_note("- Introns enriched >10 fold relative to upstream sequence but only ca 5 fold relative to downstream sequence")`

``` {r }
RSeQCdist <- read.delim("/Users/heo3/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.dist.txt", header=TRUE, stringsAsFactors=FALSE, skip=4, sep='')
RSeQCdist <- RSeQCdist[-11,]
RSeQCdist[,2:4] <- sapply(RSeQCdist[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))
knitr::kable(
  RSeQCdist
)
```

## Strand

`r margin_note("- Data consistent with second-strand experiment")`

```{r }
expt <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.expt.txt", skip=3, header=FALSE, sep=':')
colnames(expt)<-c('Class', 'Proportion')
knitr::kable(
  expt
)
```

## Insert Size Distribution

`r margin_note("- Distribution centered on zero")`
`r margin_note("- Less than 2000 of 1,000,000 reads used")`
`r margin_note("- The problem isn't the strand, so it must be the size range, but this is including a wide range of sizes")`

```{r include=FALSE}
inner_distance <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.inner_distance_freq.txt", header=FALSE)
sum(inner_distance$V3)
```

```{r dev.args = list(bg = 'transparent')}
ggplot(inner_distance, aes(x=(V1+V2)/2, y=V3)) +
  geom_bar(stat='identity') +
  ylab('reads') +
  xlab('inner distance') +
  tufte_theme()

```

## Transcript Read Coverage

`r margin_note("- Looks good. Steep drop-offs at the ends, especially 5'")`

```{r dev.args = list(bg = 'transparent')}
geneBodyCoverage <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.geneBodyCoverage.txt", header=FALSE)
geneBodyCoverage <- t(geneBodyCoverage[,-1])
ggplot(as.data.frame(geneBodyCoverage), aes(x=V1, y=V2)) +
  geom_line() +
  ylab('reads') +
  xlab('transcript percentile') +
  tufte_theme()
```

## Sequence Duplication

`r margin_note("- Plenty of duplication")`
`r margin_note("- Probably not unexpected in this case")`

```{r include=FALSE}
eval(parse(file = "~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.DupRate_plot.r")[2:6])
duplication<-rbind(
  data.frame(Reads=pos_uniqRead, Duplication_level=pos_occ, Method='Position-based'),
  data.frame(Reads=seq_uniqRead, Duplication_level=seq_occ, Method='Sequence-based')
)
```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
p<-ggplot(duplication, aes(x=Duplication_level, y=Reads, colour=Method)) +
  geom_point() +
  xlim(c(0,500)) +
  scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.8))

p+scale_colour_brewer(type = "qual", palette = 6)

```

## Splice Junctions

`r margin_note("- Events are number of transcripts")`
`r margin_note("- Junctions are number of exons")`
`r margin_note("- VAST majority of splice junctions are known")`
`r margin_note("- A lot more novel transcripts than novel splice variants")`

```{r }
junctions <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.junction.txt", skip=4, header=FALSE, sep=':')
colnames(junctions)<-c('Class', 'Number')
#junctions$Number <- formatC(as.numeric(junctions$Number), format='d', big.mark=',')
knitr::kable(
  junctions[1:8,]
)
```

## Splice Junction Saturation

`r margin_note("- It looks like both known and novel splice sites continue to be found")`

```{r include=FALSE}
eval(parse(file = "~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.junctionSaturation_plot.r")[2:5])
saturation<-rbind(
  data.frame(percent_reads=x, junctions=z, Category='All'),
  data.frame(percent_reads=x, junctions=y, Category='Known'),
  data.frame(percent_reads=x, junctions=w, Category='Novel')
)
```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
p<-ggplot(saturation, aes(y=junctions, x=percent_reads, colour=Category)) +
  geom_point() +
  #xlim(c(0,500)) +
  #scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.7))

p+scale_colour_brewer(type = "qual", palette = 6)

```

## Mutation Distribution

```{r include=FALSE}

eval(parse(file = "~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.insertion_profile.r")[c(3,9)])
eval(parse(file = "~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.deletion_profile.r")[3])
eval(parse(file = "~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/15533_300_secondstrand.mismatch_profile.r")[1:12])

SNP_types<- data.frame(Position=rep(1:100, 12), num_SNPs=c(A2C, A2G, A2T, C2A, C2G, C2T, G2A, G2C, G2T, T2A, T2C, T2G), 
                       Mutation=c(rep('A2C', 100), rep('A2G', 100), rep('A2T', 100), rep('C2A', 100), rep('C2G', 100), rep('C2T', 100), rep('G2A', 100), rep('G2C', 100), rep('G2T', 100), rep('T2A', 100), rep('T2C', 100), rep('T2G', 100)), 
                       NonDirectional=c(rep('AC', 100), rep('AG', 100), rep('AT', 100), rep('CA', 100), rep('CG', 100), rep('CT', 100), rep('CT', 100), rep('CG', 100), rep('CA', 100), rep('AT', 100), rep('CT', 100), rep('AC', 100)), 
                       SNPclass=c(rep('Transversion', 100), rep('Transition', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transition', 100), rep('Transition', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transition', 100), rep('Transversion', 100)),
                       PositionGroup = rep(c('position 1', rep('positions 2-5', 4), rep('positions 6-95', 90), rep('positions 96-99', 4), 'position 100'), 12) 
                       )

SNPcount <- group_by(SNP_types, Position) %>% summarise(Number = sum(num_SNPs)) 

mutations <- data.frame(Position=rep(1:100, 4), Number=c(SNPcount$Number, r1_insert_count, r2_insert_count, value), Type=factor(c(rep('SNP', 100), rep('Read 1 insertion',100), rep('Read 2 insertion',100), rep('Deletion', 100))))


SNPspectrum <- SNP_types %>% group_by(Mutation, PositionGroup, SNPclass) %>% summarise(Number=sum(num_SNPs))
SNPspectrum <-rbind(SNPspectrum, SNP_types %>% group_by(Mutation, SNPclass) %>% summarise(Number=sum(num_SNPs), PositionGroup = 'All'))
SNPspectrum$Bases <- ifelse(SNPspectrum$PositionGroup == 'position 1' | SNPspectrum$PositionGroup == 'position 100', 1, 
                            ifelse(SNPspectrum$PositionGroup == 'positions 2-5' | SNPspectrum$PositionGroup == 'positions 96-99', 4,
                                   ifelse(SNPspectrum$PositionGroup == 'positions 6-95', 90, 100
                                     
                                        )
                                  )
                            )
SNPspectrum$PositionGroup <- factor(SNPspectrum$PositionGroup, levels=c('position 1', 'positions 2-5', 'positions 6-95', 'positions 96-99', 'position 100', 'All'))
SNPspectrum$Mutation <- factor(SNPspectrum$Mutation, levels=c('A2G', 'T2C', 'G2A', 'C2T', 'A2C', 'T2G', 'A2T', 'T2A', 'C2G', 'G2C', 'G2T', 'C2A'))
```


`r margin_note("- For some reason, mutations are only counted separately on reads 1 and 2 for insertions")`

`r margin_note("- Approximately equal numbers of insertions, deletions and SNPs, except at the starts and end where indels drop to zero and SNPs increase")`
`r margin_note("- Odd zig-zag pattern for SNPs for some reason")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(mutations, aes(x=Position, y=Number, colour=Type)) + 
  geom_line() + 
  tufte_theme() +
  scale_colour_brewer(type = "qual", palette = 6)  + 
  theme(legend.position=c(.8,.7)) + 
  scale_y_log10()
```

`r margin_note("- Because reads 1 and 2 are mixed, equal numbers for symetric mutations")`
`r margin_note("- A2G/T2C and A2C/T2G mutations dominate at the very ends of reads, with the former dominating at the start and the latter at the end")`
`r margin_note("- These classes are also disproportionate throughout the reads, with G2A/C2T (transitions) also being higher than other (transversion) mutations")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(SNPspectrum, aes(x=Mutation, y=Number/Bases, fill=SNPclass)) + 
  geom_bar(stat='identity') + 
  tufte_theme() + 
  facet_wrap(~ PositionGroup, scales='free_y') + 
  theme(axis.text.x=element_text(angle=90)) + 
  scale_fill_brewer(type = "qual", palette = 6) + 
  theme(legend.position=c(.95,.95))
```


# Picard RNAseq Stats

`r margin_note("- Almost all reads aligned and 99\\% are the correct (second) strand")`
`r margin_note("- 9\\% of reads are intergenic, 42\\% are intronic, and 49\\% are expressed (21\\% coding, 28\\% UTR)")`

```{r }
RnaSeqMetrics <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300_secondstrand/RnaSeqMetrics.txt", header=TRUE, comment.char="#", stringsAsFactors = FALSE)
PicardStats <- gather(RnaSeqMetrics[(1),], stat, result)
PicardStats$result<-sapply(PicardStats$result, function(x) format(as.numeric(x), big.mark=','))
knitr::kable(
  PicardStats
)
```

## Distribution of Reads Among Features
`r margin_note("- VERY flat from 20% to 80%")`
`r margin_note("- Less rounded than the plot from RSeQC")`
`r margin_note("- This maybe has something to do with the coverage normalisation?")`

```{r dev.args = list(bg = 'transparent')}
cov <- data.frame(normalized_position=as.numeric(RnaSeqMetrics[-(1:2),]$PF_BASES), normalized_coverage=as.numeric(RnaSeqMetrics[-(1:2),]$PF_ALIGNED_BASES))
ggplot(cov, aes(x=normalized_position, y=normalized_coverage)) + 
  geom_line() +
  tufte_theme()
```