---
title: "Data QC"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(RColorBrewer)
library(tufte)
rm(list=ls())
source("~/BTSync/Code/R/FormatGGplot.R")
folder <- "/Users/heo3/BTSync/FetalRNAseq/BamQC/Trimmed/15533_300_trim2"
folder2 <- "/Users/heo3/BTSync/FetalRNAseq/BamQC/Trimmed/15533_trim2"
folder3 <- "/Users/heo3/BTSync/FetalRNAseq/BamQC/Trimmed/15533_1000_trim2"
```

# RSeQC
## Mapping Stats
`r margin_note("- Excludes reads mapping to rDNA")`
`r margin_note("- Total records = mapped reads + non-primary")`
`r margin_note("- Number of uniquely mapped reads goes up slightly with inner distance")`
`r margin_note("- Number of properly paired reads increases considerably")`
`r margin_note("- Number of non-primary hits peaks at 300 bp")`
`r margin_note("- Number of non-primary hits down dramatically from 194 million and the number of uniquely mapped reads has gone up from 226 million")`

``` {r }
RSeQCstats <- read.delim(paste(folder2, ".ex.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats[4,2] <-strsplit(RSeQCstats[4,1], ' +')[[1]][4]
RSeQCstats[4,1] <- 'Non primary hits'
RSeQCstats[,2] <- formatC(as.numeric(RSeQCstats[,2]), format='d', big.mark=',')
RSeQCstats2 <- read.delim(paste(folder, ".ex.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats2[4,2] <-strsplit(RSeQCstats2[4,1], ' +')[[1]][4]
RSeQCstats2[,2] <- formatC(as.numeric(RSeQCstats2[,2]), format='d', big.mark=',')
RSeQCstats <- cbind(RSeQCstats, RSeQCstats2[,2])
RSeQCstats3 <- read.delim(paste(folder3, ".ex.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats3[4,2] <-strsplit(RSeQCstats3[4,1], ' +')[[1]][4]
RSeQCstats3[,2] <- formatC(as.numeric(RSeQCstats3[,2]), format='d', big.mark=',')
RSeQCstats <- cbind(RSeQCstats, RSeQCstats3[,2])
colnames(RSeQCstats)<-c('stat', '50bp', '300bp', '1000bp')
knitr::kable(
  RSeQCstats
)
```

## rRNA Mapping Stats

`r margin_note("- 14 million reads map to rDNA (ca. 5%)")`
`r margin_note("- Number of mappings is considerably higher for 300 bp for some reason")`
`r margin_note("- These mappings account for 45% of multimapped reads and 70% of all non-primary hits")`
`r margin_note("- These numbers are up from 20% and 33% respectively")`

``` {r }
RSeQCstats <- read.delim(paste(folder2, ".in.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats[4,2] <-strsplit(RSeQCstats[4,1], ' +')[[1]][4]
RSeQCstats[4,1] <- 'Non primary hits'
RSeQCstats[,2] <- formatC(as.numeric(RSeQCstats[,2]), format='d', big.mark=',')
RSeQCstats2 <- read.delim(paste(folder, ".in.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats2[4,2] <-strsplit(RSeQCstats2[4,1], ' +')[[1]][4]
RSeQCstats2[,2] <- formatC(as.numeric(RSeQCstats2[,2]), format='d', big.mark=',')
RSeQCstats <- cbind(RSeQCstats, RSeQCstats2[,2])
RSeQCstats3 <- read.delim(paste(folder3, ".in.stats.txt", sep=""), header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':')
RSeQCstats3[4,2] <-strsplit(RSeQCstats3[4,1], ' +')[[1]][4]
RSeQCstats3[,2] <- formatC(as.numeric(RSeQCstats3[,2]), format='d', big.mark=',')
RSeQCstats <- cbind(RSeQCstats, RSeQCstats3[,2])
colnames(RSeQCstats)<-c('stat', '50bp', '300bp', '1000bp')
knitr::kable(
  RSeQCstats
)
```


## Distribution of Reads Among Features (tags per kb)
`r margin_note("- Exons enriched >10 fold relative to introns")`
`r margin_note("- Introns enriched >10 fold relative to upstream sequence but only ca 5 fold relative to downstream sequence")`
`r margin_note("- These numbers are EXACTLY the same")`

``` {r }
RSeQCdist <- read.delim(paste(folder2, ".dist.txt", sep=""), header=TRUE, stringsAsFactors=FALSE, skip=4, sep='')
RSeQCdist <- RSeQCdist[-11,]
RSeQCdist[,2:4] <- sapply(RSeQCdist[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))
RSeQCdist2 <- read.delim(paste(folder, ".dist.txt", sep=""), header=TRUE, stringsAsFactors=FALSE, skip=4, sep='')
RSeQCdist2 <- RSeQCdist2[-11,]
RSeQCdist2[,2:4] <- sapply(RSeQCdist2[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))
RSeQCdist3 <- read.delim(paste(folder3, ".dist.txt", sep=""), header=TRUE, stringsAsFactors=FALSE, skip=4, sep='')
RSeQCdist3 <- RSeQCdist3[-11,]
RSeQCdist3[,2:4] <- sapply(RSeQCdist3[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))
RSeQCdist <- cbind(RSeQCdist[,c(1,4)], RSeQCdist2[,4], RSeQCdist3[,4])
colnames(RSeQCdist)<-c('Group', '50bp', '300bp', '1000bp')
knitr::kable(
  RSeQCdist
)
```

## Strand

`r margin_note("- Data consistent with second-strand experiment")`
`r margin_note("- The proportion that can't be determined is up slightely from 0.25")`

```{r }
expt <- cbind(read.delim(paste(folder2, ".expt.txt", sep=""), skip=3, header=FALSE, sep=':'),
             read.delim(paste(folder, ".expt.txt", sep=""), skip=3, header=FALSE, sep=':')[,2],
             read.delim(paste(folder3, ".expt.txt", sep=""), skip=3, header=FALSE, sep=':')[,2]
)
colnames(expt)<-c('Class', '50bp', '300bp', '1000bp')
knitr::kable(
  expt
)
```

## Insert Size Distribution

`r margin_note("- Distribution centered on zero")`
`r margin_note("- 979,000 of 1,000,000 reads used (up from <2000)")`
`r margin_note("- Shape is unchanged, but a lot smoother")`

```{r include=FALSE}
inner_distance <- cbind(read.delim(paste(folder2, ".inner_distance_freq.txt", sep=""), header=FALSE),
                        read.delim(paste(folder, ".inner_distance_freq.txt", sep=""), header=FALSE)[,3],
                        read.delim(paste(folder3, ".inner_distance_freq.txt", sep=""), header=FALSE)[,3]
)
colnames(inner_distance) <- c('V1', 'V2', '50bp', '300bp', '1000bp')
inner_distance[,3:5] %>% summarise_each(funs(sum))
inner_distance<- gather(inner_distance, distance, V3, 3:5)
```

```{r dev.args = list(bg = 'transparent')}
ggplot(inner_distance, aes(x=(V1+V2)/2, y=V3, colour=distance)) +
  geom_line() +
  ylab('reads') +
  xlab('inner distance') +
  tufte_theme() +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6)

```

## Transcript Read Coverage

`r margin_note("- Looks good. Steep drop-offs at the ends, especially 5'")`
`r margin_note("- Insert size makes no difference")`

```{r dev.args = list(bg = 'transparent')}
geneBodyCoverage <- read.delim(paste(folder2, ".geneBodyCoverage.txt", sep=""), header=FALSE)
geneBodyCoverage <- as.data.frame(t(geneBodyCoverage[,-1]))
geneBodyCoverage$V3 <- '50bp'
geneBodyCoverage2 <- read.delim(paste(folder, ".geneBodyCoverage.txt", sep=""), header=FALSE)
geneBodyCoverage2 <- as.data.frame(t(geneBodyCoverage2[,-1]))
geneBodyCoverage2$V3 <- '300bp'
geneBodyCoverage3 <- read.delim(paste(folder3, ".geneBodyCoverage.txt", sep=""), header=FALSE)
geneBodyCoverage3 <- as.data.frame(t(geneBodyCoverage3[,-1]))
geneBodyCoverage3$V3 <- '1000bp'
geneBodyCoverage <- rbind(geneBodyCoverage, geneBodyCoverage2, geneBodyCoverage3)
ggplot(geneBodyCoverage, aes(x=V1, y=V2, colour=V3)) +
  geom_line() +
  ylab('reads') +
  tufte_theme() +
  xlab('transcript percentile') +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6)
  
```

## Sequence Duplication

`r margin_note("- Plenty of duplication")`
`r margin_note("- Probably not unexpected in this case")`

```{r include=FALSE}
eval(parse(file = paste(folder, ".DupRate_plot.r", sep=""))[2:6])
duplication<-rbind(
  data.frame(Reads=pos_uniqRead, Duplication_level=pos_occ, Method='Position-based'),
  data.frame(Reads=seq_uniqRead, Duplication_level=seq_occ, Method='Sequence-based')
)
```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
p<-ggplot(duplication, aes(x=Duplication_level, y=Reads, colour=Method)) +
  geom_point() +
  xlim(c(0,500)) +
  scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.8))

p+scale_colour_brewer(type = "qual", palette = 6)

```

## Splice Junctions

`r margin_note("- Events are number of transcripts")`
`r margin_note("- Junctions are number of exons")`
`r margin_note("- VAST majority of splice junctions are known")`
`r margin_note("- A lot more novel transcripts with smaller insert size")`

```{r }
junctions <- read.delim(paste(folder2, ".junction.txt", sep=""), skip=4, header=FALSE, sep=':')
junctions <-cbind(junctions, read.delim(paste(folder, ".junction.txt", sep=""), skip=4, header=FALSE, sep=':')[,2])
junctions <-cbind(junctions, read.delim(paste(folder3, ".junction.txt", sep=""), skip=4, header=FALSE, sep=':')[,2])
junctions[,2:4] <- sapply(junctions[,2:4], function(x) formatC(as.numeric(x), format='d', big.mark=','))

colnames(junctions)<-c('Class', '50bp', '300bp', '1000bp')

knitr::kable(
  junctions[1:8,]
)
```

## Splice Junction Saturation

`r margin_note("- It looks like both known and novel splice sites continue to be found")`

```{r include=FALSE}
eval(parse(file = paste(folder, ".junctionSaturation_plot.r", sep=""))[2:5])
saturation<-rbind(
  data.frame(percent_reads=x, junctions=z, Category='All'),
  data.frame(percent_reads=x, junctions=y, Category='Known'),
  data.frame(percent_reads=x, junctions=w, Category='Novel')
)
```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
p<-ggplot(saturation, aes(y=junctions, x=percent_reads, colour=Category)) +
  geom_point() +
  #xlim(c(0,500)) +
  #scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.7))

p+scale_colour_brewer(type = "qual", palette = 6)

```

## Mutation Distribution

```{r include=FALSE}

eval(parse(file = paste(folder, ".insertion_profile.r", sep=""))[c(3,9)])
eval(parse(file =paste(folder, ".deletion_profile.r", sep=""))[3])
eval(parse(file =paste(folder, ".mismatch_profile.r", sep=""))[1:12])

SNP_types<- data.frame(Position=rep(1:100, 12), num_SNPs=c(A2C, A2G, A2T, C2A, C2G, C2T, G2A, G2C, G2T, T2A, T2C, T2G), 
                       Mutation=c(rep('A2C', 100), rep('A2G', 100), rep('A2T', 100), rep('C2A', 100), rep('C2G', 100), rep('C2T', 100), rep('G2A', 100), rep('G2C', 100), rep('G2T', 100), rep('T2A', 100), rep('T2C', 100), rep('T2G', 100)), 
                       NonDirectional=c(rep('AC', 100), rep('AG', 100), rep('AT', 100), rep('CA', 100), rep('CG', 100), rep('CT', 100), rep('CT', 100), rep('CG', 100), rep('CA', 100), rep('AT', 100), rep('CT', 100), rep('AC', 100)), 
                       SNPclass=c(rep('Transversion', 100), rep('Transition', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transition', 100), rep('Transition', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transversion', 100), rep('Transition', 100), rep('Transversion', 100)),
                      PositionGroup = rep(c('position 1', rep('positions 2-5', 4), rep('positions 6-95', 90), rep('positions 96-99', 4), 'position 100'), 12) 
                     )

SNPcount <- group_by(SNP_types, Position) %>% summarise(Number = sum(num_SNPs)) 

#mutations <- data.frame(Position=rep(1:100, 4), Number=c(SNPcount$Number, r1_insert_count, r2_insert_count, value), Type=factor(c(rep('SNP', 100), rep('Read 1 insertion',100), rep('Read 2 insertion',100), rep('Deletion', 100))))
mutations <- data.frame(Position=rep(1:100, 4), Number=c(SNPcount$Number, value), Type=factor(c(rep('SNP', 100), rep('Deletion', 100))))


SNPspectrum <- SNP_types %>% group_by(Mutation, PositionGroup, SNPclass) %>% summarise(Number=sum(num_SNPs))
SNPspectrum <-rbind(SNPspectrum, SNP_types %>% group_by(Mutation, SNPclass) %>% summarise(Number=sum(num_SNPs), PositionGroup = 'All'))
SNPspectrum$Bases <- ifelse(SNPspectrum$PositionGroup == 'position 1' | SNPspectrum$PositionGroup == 'position 100', 1, 
                            ifelse(SNPspectrum$PositionGroup == 'positions 2-5' | SNPspectrum$PositionGroup == 'positions 96-99', 4,
                                   ifelse(SNPspectrum$PositionGroup == 'positions 6-95', 90, 100
                                     
                                        )
                                  )
                            )
SNPspectrum$PositionGroup <- factor(SNPspectrum$PositionGroup, levels=c('position 1', 'positions 2-5', 'positions 6-95', 'positions 96-99', 'position 100', 'All'))
SNPspectrum$Mutation <- factor(SNPspectrum$Mutation, levels=c('A2G', 'T2C', 'G2A', 'C2T', 'A2C', 'T2G', 'A2T', 'T2A', 'C2G', 'G2C', 'G2T', 'C2A'))
```


`r margin_note("- For some reason, mutations are only counted separately on reads 1 and 2 for insertions")`

`r margin_note("- Approximately equal numbers of insertions, deletions and SNPs, except at the starts and end where indels drop to zero and SNPs increase")`
`r margin_note("- Odd zig-zag pattern for SNPs for some reason")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(mutations, aes(x=Position, y=Number, colour=Type)) + 
  geom_line() + 
  tufte_theme() +
  scale_colour_brewer(type = "qual", palette = 6)  + 
  theme(legend.position=c(.8,.7)) + 
  scale_y_log10()
```

`r margin_note("- Because reads 1 and 2 are mixed, equal numbers for symetric mutations")`
`r margin_note("- A2G/T2C and A2C/T2G mutations dominate at the very ends of reads, with the former dominating at the start and the latter at the end")`
`r margin_note("- These classes are also disproportionate throughout the reads, with G2A/C2T (transitions) also being higher than other (transversion) mutations")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(SNPspectrum, aes(x=Mutation, y=Number/Bases, fill=SNPclass)) + 
  geom_bar(stat='identity') + 
  tufte_theme() + 
  facet_wrap(~ PositionGroup, scales='free_y') + 
  theme(axis.text.x=element_text(angle=90)) + 
  scale_fill_brewer(type = "qual", palette = 6) + 
  theme(legend.position=c(.95,.95))
```
