---
title: "Data QC"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(RColorBrewer)
library(tufte)
rm(list=ls())
source("~/BTSync/Code/R/FormatGGplot.R")
folders <- c("/Users/heo3/BTSync/FetalRNAseq/BamQC/15533/15533",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16438/16438",
             # "/Users/heo3/BTSync/FetalRNAseq/BamQC/16972/16972",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17087/17087",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17109/17109",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/17701/17701",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18294/18294",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18655/18655",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/15768/15768",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16286/16286",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17054/17054",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17068/17068",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17081/17081",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17115/17115",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17130/17130",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17812/17812",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18349/18349",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18596/18596",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/15468/15468",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16488/16488",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16840/16840",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16929/16929",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17049/17049",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18687/18687",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/15641/15641",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16024/16024",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16115/16115",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16385/16385",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16428/16428",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16491/16491",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16548/16548",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16810/16810",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/16826/16826",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17048/17048",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/17053/17053",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17071/17071",
             # no gene body coverage "/Users/heo3/BTSync/FetalRNAseq/BamQC/17921-l1/17921-l1",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/15655/15655",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16483/16483",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16640/16640",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17013/17013",
             # < 2000 reads "/Users/heo3/BTSync/FetalRNAseq/BamQC/17025/17025",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17198/17198",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17229/17229",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17333/17333",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17475/17475",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17543/17543",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17629/17629",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17835/17835",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18249/18249", 
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18372/18372",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18666/18666",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18983/18983",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/19043/19043",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/19052/19052",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/A138/A138",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/A226/A226",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/15240/15240",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/16649/16649",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17072/17072",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17160/17160",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17167/17167",
             #"/Users/heo3/BTSync/FetalRNAseq/BamQC/17175/17175",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17671/17671",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17922/17922",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/17923/17923",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18121/18121",
             "/Users/heo3/BTSync/FetalRNAseq/BamQC/18134/18134"

             )
LibraryInfo <- read.delim("/Users/heo3/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt", 
                     header=TRUE, stringsAsFactors=TRUE, sep='\t'
                     )
```

# RSeQC
## Mapping Stats
`r margin_note("- rDNA reads not included in first three stats")`
`r margin_note("- 12 samples have < 10% rRNA reads")`
`r margin_note("- All samples have < 20% except for 17812 (24%), 17054 (44%) and 18655 (50%)")`

``` {r }
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".ex.stats.txt", sep=""), 
                     header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                     )
  temp[4,2] <-strsplit(temp[4,1], ' +')[[1]][4]
  temp[4,1] <- 'Non primary hits'
  temp <- temp[c(6,7,14),]
  temp2 <- read.delim(paste(folder, ".in.stats.txt", sep=""), 
                     header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                     )
  temp2[4,2] <-strsplit(temp[4,1], ' +')[[1]][4]
  temp2[4,1] <- 'Non primary hits'
  temp <-rbind(temp, c("rDNA",sum(as.numeric(temp2[c(6,7),2]))))
  temp[,1] <- c("Multimapped", "Unique", "Paired", "rDNA")
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)
#mutate(RSeQCstats, Perc_rDNA = as.numeric(rDNA)/(as.numeric(Unique)+as.numeric(Multimapped)+as.numeric(rDNA))) %>% arrange(Perc_rDNA)
ReadNumbers <- RSeQCstats[,c(1,5)]
RSeQCstats <- RSeQCstats[c(1,5,3,2,4)]
RSeQCstats[,2:5] <- sapply(RSeQCstats[,2:5], function(x) formatC(as.numeric(x), format='d', big.mark=','))
knitr::kable(
  RSeQCstats
)
```


## Distribution of Reads Among Features (tags per kb per million reads)
`r margin_note("- Intergenic only considers sequence within 10 kb upstream of TSS and downstream of TES")`
`r margin_note("- Exons enriched >10 fold relative to introns")`
`r margin_note("- Introns enriched >10 fold relative to Intergenic")`


``` {r }
RSeQCstats <- data.frame()
for (folder in folders) {
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp <- read.delim(paste(folder, ".dist.txt", sep=""), 
                           header=TRUE, stringsAsFactors=FALSE, skip=4, sep=''
                     )
  temp <- temp[-11,]
  temp <- data.frame(V1 = c("CDS", "UTR", "Intron", "Intergenic"), 
             V2=c(temp[1,4]/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)*1000000, 
                  sum(as.numeric(temp[c(2,3),3]))/sum(as.numeric(temp[c(2,3),2]))*1000/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)*1000000,
                  temp[4,4]/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)*1000000,
                  sum(as.numeric(temp[c(7,10),3]))/sum(as.numeric(temp[c(7,10),2]))*1000/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)*1000000
                  )
             )
  temp$sample <- sample
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)
RSeQCstats <- RSeQCstats[c(1,2,5,4,3)]

knitr::kable(
  RSeQCstats
)
```

## Strand

`r margin_note("- Data consistent with second-strand experiment")`

```{r }
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".expt.txt", sep=""), 
                     skip=3, header=FALSE, sep=':'
                     )
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$V1 <- c("Ambiguous", "First Strand", "Second Strand")
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)

knitr::kable(
  RSeQCstats
)
```

## Insert Size Distribution

`r margin_note("- Inner distance pretty consistenly at - 125 for Edinbrugh data, but centered at about -50 for  Exeter data")`
`r margin_note("- This indicates that fragment size tends to be around 125-150 bp")`


```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".inner_distance_freq.txt", sep=""),
                     header=FALSE
                     )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCstats <- rbind(RSeQCstats, temp)
}
```

```{r dev.args = list(bg = 'transparent')}
ggplot(RSeQCstats, aes(x=(V1+V2)/2, y=V3, colour=centre, group=sample)) +
  geom_line() +
  ylab('reads') +
  xlab('inner distance') +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  tufte_theme() 

```

## Transcript Read Coverage

`r margin_note("- Looks good. Steep drop-offs at the ends, especially 5'")`
`r margin_note("- Not sure why 15533 is so much higher than all others")`

```{r dev.args = list(bg = 'transparent')}
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".geneBodyCoverage.txt", sep=""),
                     header=FALSE
                     )
  temp <-  as.data.frame(t(temp[,-1]))
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  temp$V2 <- temp$V2
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

ggplot(RSeQCstats, aes(x=V1, y=V2/ReadNumbers, colour=centre, group=sample)) +
  geom_line() +
  ylab('reads') +
  tufte_theme() +
  theme(legend.position=c(0.1,.9)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  xlab('transcript percentile') +
  ylab("proportion of uniquely mapped reads")
  
```

## Sequence Duplication

`r margin_note("- Plenty of duplication")`
`r margin_note("- Probably not unexpected in this case")`
`r margin_note("- Duplication higher for Exeter samples")`

```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".DupRate_plot.r", sep=""))[2:6])
  temp <- rbind(data.frame(Reads=pos_uniqRead, 
                            Duplication_level=pos_occ, 
                            Method='Position-based'
                            ),
                 data.frame(Reads=seq_uniqRead, 
                            Duplication_level=seq_occ, 
                            Method='Sequence-based'
                            )
                )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(RSeQCstats, aes(x=Duplication_level, y=Reads/ReadNumbers, colour=centre, group=sample)) +
  geom_point(alpha=.2) +
  xlim(c(0,500)) +
  scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  facet_wrap(~ Method) +
  ylab("Proportion of Reads")


```

## Splice Junctions

`r margin_note("- VAST majority of splice junctions are known")`

```{r }
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".junction.bed", sep=""), sep='\t', header=FALSE)
  temp <- group_by(temp, V4) %>% summarise(Number=n())
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats[,2] <- formatC(as.numeric(RSeQCstats[,2][[1]]), format='d', big.mark=',')
RSeQCstats[,1] <- RSeQCstats[,1][[1]]
RSeQCstats <- spread(RSeQCstats, V4, Number)

knitr::kable(
  RSeQCstats
)
```

## Splice Junction Saturation

`r margin_note("- It looks like both known and novel splice sites continue to be found")`

```{r include=FALSE}
RSeQCstats <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".junctionSaturation_plot.r", sep=""))[2:5])
  temp<-rbind(
              data.frame(percent_reads=x, junctions=z, Category='All'),
              data.frame(percent_reads=x, junctions=y, Category='Known'),
              data.frame(percent_reads=x, junctions=w, Category='Novel')
              )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,2])
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCstats <- rbind(RSeQCstats, temp)
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(RSeQCstats, aes(y=junctions, x=ReadNumbers*percent_reads/100, group=sample, colour=centre)) +
  geom_point() +
  #xlim(c(0,500)) +
  #scale_y_log10() +
  scale_colour_brewer(type = "qual", palette = 6) +
  tufte_theme() +
  theme(legend.position=c(.8,.9)) +
  facet_wrap(~ Category, scales='free') +
  xlab("Number of Reads")

```

## Mutation Distribution

```{r include=FALSE}
RSeQCmutations <- data.frame()
RSeQCspectrum <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".insertion_profile.r", sep=""))[c(3,9)])
  eval(parse(file =paste(folder, ".deletion_profile.r", sep=""))[3])
  eval(parse(file =paste(folder, ".mismatch_profile.r", sep=""))[1:12])
  read_len = length(A2C)
  SNP_types<- data.frame(Position=rep(1:read_len, 12), 
                         num_SNPs=c(A2C, A2G, A2T, C2A, C2G, C2T, G2A, 
                                    G2C, G2T, T2A, T2C, T2G
                                    ), 
                         Mutation=c(rep('A2C', read_len), rep('A2G', read_len), 
                                    rep('A2T', read_len), rep('C2A', read_len), 
                                    rep('C2G', read_len), rep('C2T', read_len), 
                                    rep('G2A', read_len), rep('G2C', read_len), 
                                    rep('G2T', read_len), rep('T2A', read_len), 
                                    rep('T2C', read_len), rep('T2G', read_len)
                                    ), 
                         NonDirectional=c(rep('AC', read_len), rep('AG', read_len),
                                          rep('AT', read_len), rep('CA', read_len),
                                          rep('CG', read_len), rep('CT', read_len),
                                          rep('CT', read_len), rep('CG', read_len),
                                          rep('CA', read_len), rep('AT', read_len),
                                          rep('CT', read_len), rep('AC', read_len)
                                          ), 
                         SNPclass=c(rep('Transversion', read_len), 
                                    rep('Transition', read_len), 
                                    rep('Transversion', read_len),
                                    rep('Transversion', read_len),
                                    rep('Transversion', read_len), 
                                    rep('Transition', read_len), 
                                    rep('Transition', read_len), 
                                    rep('Transversion', read_len),
                                    rep('Transversion', read_len),
                                    rep('Transversion', read_len),
                                    rep('Transition', read_len),
                                    rep('Transversion', read_len)
                                    ),
                      PositionGroup = rep(c('First position', 
                                            rep('First 5', 4), 
                                            rep('Middle', read_len - 10),
                                            rep('Last 5', 4), 'Last position'
                                            ), 
                                          12
                                          ) 
                     )

  SNPcount <- group_by(SNP_types, Position) %>% summarise(Number = sum(num_SNPs)) 

  sample <- tail(strsplit(folder, '/')[[1]], 1)
  mutations <- data.frame(Position=c(1:read_len, 
                                     1:length(r1_insert_count),
                                     1:length(r2_insert_count),
                                     1:length(value)
                                     ), 
                          Number=c(SNPcount$Number,
                                   r1_insert_count, 
                                   r2_insert_count,
                                   value
                                   ), 
                          ReadNumbers = rep(as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique), 
                                            read_len + length(r1_insert_count) + length(r2_insert_count) + length(value)),
                          Type=factor(c(rep('SNP', read_len), 
                                        rep('Read 1 insertion',
                                            length(r1_insert_count)
                                            ),
                                        rep('Read 2 insertion',
                                            length(r2_insert_count)
                                            ),
                                        rep('Deletion', 
                                            length(value)
                                            )
                                    ))
                          )

  SNPspectrum <- SNP_types %>% 
    group_by(Mutation, PositionGroup, SNPclass) %>% 
    summarise(Number=sum(num_SNPs))
  SNPspectrum <-rbind(SNPspectrum, 
                      SNP_types %>% 
                        group_by(Mutation, SNPclass) %>%
                        summarise(Number=sum(num_SNPs), 
                                  PositionGroup = 'All'
                                 )
                      )
  SNPspectrum$Bases <- ifelse(SNPspectrum$PositionGroup == 'First position' |
                                SNPspectrum$PositionGroup == 'Last position', 1, 
                              ifelse(SNPspectrum$PositionGroup == 'First 5' |
                                       SNPspectrum$PositionGroup == 'Last 5', 4,
                                     ifelse(SNPspectrum$PositionGroup == 'Middle', 
                                       read_len-10,
                                       read_len
                                      )
                                  )
                            )
  SNPspectrum$PositionGroup <- factor(SNPspectrum$PositionGroup, 
                                    levels=c('First position', 
                                             'First 5', 
                                             'Middle', 
                                             'Last 5', 
                                             'Last position',
                                             'All'
                                             )
                                    )
  SNPspectrum$Mutation <- factor(SNPspectrum$Mutation, 
                               levels=c('A2G', 'T2C', 'G2A', 'C2T', 'A2C', 'T2G',
                                        'A2T', 'T2A', 'C2G', 'G2C', 'G2T', 'C2A'
                                        )
                               )
  mutations$sample <- sample
  mutations$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCmutations <- rbind(RSeQCmutations, mutations)
  SNPspectrum$sample <- sample
  RSeQCspectrum <- rbind(RSeQCspectrum, SNPspectrum)
}

```


`r margin_note("- For some reason, mutations are only counted separately on reads 1 and 2 for insertions")`

`r margin_note("- Approximately equal numbers of insertions, deletions and SNPs, except at the starts and end where indels drop to zero and SNPs increase")`
`r margin_note("- Odd zig-zag pattern for SNPs for some reason")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE}
#ggplot(subset(RSeQCmutations, centre=='Exeter'), aes(x=Position, y=Number, group=sample, colour=sample)) + 
ggplot(RSeQCmutations, aes(x=Position, y=Number/ReadNumbers*1000000, group=sample, colour=centre)) + 
  geom_line() + 
  tufte_theme() +
  scale_colour_brewer(type = "qual", palette = 6)  + 
  theme(legend.position=c(.8,.9)) + 
  #scale_y_log10() +
  facet_wrap(~ Type, nrow=2) +
  ylab("Mutations per Million Reads")
```

`r margin_note("- Because reads 1 and 2 are mixed, equal numbers for symetric mutations")`
`r margin_note("- A2G/T2C and A2C/T2G mutations dominate at the very ends of reads, with the former dominating at the start and the latter at the end")`
`r margin_note("- These classes are also disproportionate throughout the reads, with G2A/C2T (transitions) also being higher than other (transversion) mutations")`


```{r dev.args = list(bg = 'transparent'), warning=FALSE, fig.height=20}
ggplot(RSeQCspectrum, aes(x=Mutation, y=Number/Bases, fill=SNPclass)) + 
  geom_bar(stat='identity') + 
  tufte_theme() + 
  facet_grid(sample ~ PositionGroup, scales='free_y') + 
  theme(axis.text.x=element_text(angle=90)) + 
  scale_fill_brewer(type = "qual", palette = 6) + 
  theme(legend.position=c(.95,.95))
```
