---
title: "Data QC"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
```
## Insert Sizes
- 20.7% of reads have an insert size of zero
- 10.9% have an insert size > 5000 bp
- BIG jump right at 1000 bp, stays high until 1750, then gradually declines

```{r}

insert_sizes <- read.delim("/Users/heo3/BTSync/FetalRNAseq/BamQC/15533_300/insert_sizes.txt", header=TRUE, stringsAsFactors=FALSE)
colnames(insert_sizes)<-c('size', 'percent')
sum(insert_sizes[insert_sizes$size > 5000, 2])
insert_sizes[1,2]
ggplot(insert_sizes[insert_sizes$size > 0 & insert_sizes$size < 5001,], aes(x=size, y=percent)) +geom_line()

```

## Picard RNAseq Stats
- Almost all reads aligned and 99% are the correct (second) strand
- 15% of reads are intergenic, 45% are intronic, and 40% are expressed (22% coding, 18% UTR)
- NO reads map to rRNA (this seems like it must be an error)
- The feature coverage plot looks good. Steep drop-offs at the ends, and a slight bias toward 3' (this could be due to active transcription)


```{r }
RnaSeqMetrics <- read.delim("~/BTSync/FetalRNAseq/BamQC/15533_300/RnaSeqMetrics.txt", header=TRUE, comment.char="#", stringsAsFactors = FALSE)

cov <- data.frame(normalized_position=as.numeric(RnaSeqMetrics[-(1:2),]$PF_BASES), normalized_coverage=as.numeric(RnaSeqMetrics[-(1:2),]$PF_ALIGNED_BASES))
RnaSeqMetrics <- RnaSeqMetrics[(1),]
ggplot(cov, aes(x=normalized_position, y=normalized_coverage))+geom_bar(stat='identity')
gather(RnaSeqMetrics, stat, result)
```

# RSeQC

==================================================
All numbers are READ count
==================================================

- Total records:                          374696825

- QC failed:                              0
- Optical/PCR duplicate:                  0
- Non primary hits                        91094701
- Unmapped reads:                         0
- mapq < mapq_cut (non-unique):           38969798

- mapq >= mapq_cut (unique):              244632326
- Read-1:                                 123273973
- Read-2:                                 121358050
- Reads map to '+':                       122285034
- Reads map to '-':                       122347292
- Non-splice reads:                       213432309
- Splice reads:                           31200017
- Reads mapped in proper pairs:           1285466
- Proper-paired reads map to different chrom:1316

