---
title: "Cufflinks"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
rm(list=ls())
library(dplyr)
library(data.table)
library(ggplot2)
library(tidyr)
library(tufte)
library(RColorBrewer)
library(scales)
source("~/BTSync/Code/R/FormatGGplot.R")

```


```{r include=FALSE}
Combined <- fread("~/BTSync/FetalRNAseq/Cufflinks/Combined.tracking", header=FALSE)
colnames(Combined) <- c("transfrag_id", 
                     "locus_id", 
                     "gene_id",                     
                     "class",
                     "15468",
                     "15533",
                     "15768",
                     "16286",
                     "16438",
                     "16488",
                     "16840",
                     "16929",
                     "16972",
                     "17049",
                     "17054",
                     "17068",
                     "17081",
                     "17087",
                     "17109",
                     "17115",
                     "17130",
                     "17701",
                     "17812",
                     "18294",
                     "18349",
                     "18596",
                     "18655",
                     "18687"
                     )
AllTranscripts <- ifelse(Combined[,5:28, with=FALSE] == '-', 0, 1)
Combined$Count<- rowSums(ifelse(Combined[,5:28, with=FALSE] == '-', 0, 1))

```

# Cufflinks
## Transcript saturation (mean of 100 random samples) 
`r margin_note("- 207,000 match reference (=)")`
`r margin_note("- 216,000 intronic (i)")`
`r margin_note("- 77,000 fall in repeat masked regions (r)")`
`r margin_note("- 51,000 multiple classes (.)")`
`r margin_note("- 48,000 intergenic (u)")`
`r margin_note("- 36,000 splice variants (j)")`
`r margin_note("- 32,000 antisense (x)")`
`r margin_note("- 5000 pre-mRNA (e)")`
`r margin_note("- 4000 polymerse run-on (p)")`
`r margin_note("- 1000 generic exon overlap (o)")`
`r margin_note("- 89 contained (c)")`
`r margin_note("- 27 intron antisense (s)")`


```{r pressure, echo=FALSE}
#knitr::kable(
#group_by(Combined, class) %>% summarise(n())
#)
```


``` {r include=FALSE}
SampleRows <- function(x, n) {
  if ( n ==1 ) {
  counts <- x[, sample(ncol(x), size = n)]  
  }
  else {
  counts <- x[, sample(ncol(x), size = n)] %>% rowSums() 
  }
  length(counts[counts > 0])  
}

SampleMean <- function(data, n, reps) {
  countList <- vector(,reps)
  for (x in 1:reps){
    countList[x] <- SampleRows(data, n)
  }
  mean(countList)
}

RarefyTranscripts <- function(Subset, reps) {
  Means <- vector(,ncol(Subset))
  for (x in 1:ncol(Subset)) {
    Means[x] <- SampleMean(Subset, x, reps)
  }
  data.frame(N = 1:ncol(Subset), Mean=Means)
}
#SatData <- data.frame()
#for ( trans_class in unique(select(Combined, class))[[1]]) {
  #print(trans_class)
#  Subset <- ifelse(filter(Combined, class == trans_class)[,5:26, with=F] == '-', 0, 1)
  #print(nrow(Subset))
#  temp <- RarefyTranscripts(Subset, 100)
#  temp$Class <- trans_class
#  SatData <-rbind(SatData, temp)
#}
#dput(SatData, file="~/BTSync/FetalRNAseq/Cufflinks/Saturation.r")
SatData <- dget(file="~/BTSync/FetalRNAseq/Cufflinks/Saturation.r")
```

`r margin_note("- 'multiple classes' is the only class that shows evidence of saturation")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE, fig.height=8}
ggplot(SatData, aes(x=N, y=Mean, colour=Class)) + 
  geom_line() + 
  tufte_theme() +
  theme(legend.position=c(.9,.65)) 
 
```

`r margin_note("- all intergenic features and almost all intron antisense features are singletons")`
`r margin_note("- 'multiple classes' is bimodal, with most being confined to 2 samples or present in all")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Combined, aes(x=Count, fill=class))+geom_bar(position='dodge') +
  facet_grid(class ~ ., scales="free_y") + 
  tufte_theme() +
  scale_y_continuous(breaks=pretty_breaks(n=2), labels=comma)
```

`r margin_note("- Y-axes scaled the same. Most abundant classes excluded")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Combined, aes(x=Count, fill=class))+geom_bar(position='dodge') +
  facet_grid(class ~ .) + 
  scale_y_continuous(limits=c(0,10000), breaks=c(0,5000,10000)) +
  tufte_theme()

```

# Mixed category
`r margin_note("- intronic, repeats, intergenic and antisense are the most common types in the 'multiple types' category")`
`r margin_note("- ca. 300 reference transcripts that overlap an additional feature in another category")`
```{r include=FALSE}
Mixed <- read.delim("~/BTSync/FetalRNAseq/Cufflinks/Mixed.txt", header=FALSE)

colnames(Mixed) <- c('transfrag_id', '=', 'i', 'r', 'u', 'j', 'x', 'e', 'p', 'o', 'c', 's')
summarise_each(Mixed[,-1], funs(sum(.)))
Mixed$total <- rowSums(Mixed[,-1])
Mixed2 <- gather(Mixed, class, count, -transfrag_id, -total)

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Mixed2, aes(x=count, fill=class)) +
  geom_bar(position='dodge') +
  facet_grid(class ~ ., scales="free_y") +
  tufte_theme() +
  scale_x_continuous(limits=c(.5,25)) +
  scale_y_continuous(breaks=pretty_breaks(n=2)) +
  scale_fill_manual(values=gg_color_hue(12)[2:12])
```

`r margin_note("- Y-axes scaled the same. Most abundant classes excluded")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Mixed2, aes(x=count, fill=class)) +
  geom_bar(position='dodge') +
  facet_grid(class ~ .) +
  scale_x_continuous(limits=c(.5,25)) +
  scale_y_continuous(limits=c(0,2000), breaks=c(0,1000,2000)) +
  tufte_theme() +
  scale_fill_manual(values=gg_color_hue(12)[2:12])


```
