---
title: "Cufflinks"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
rm(list=ls())
library(dplyr)
#library(data.table)
library(ggplot2)
library(tidyr)
library(tufte)
library(RColorBrewer)
library(scales)
source("~/BTSync/Code/R/FormatGGplot.R")
library(RMySQL)

```


```{r include=FALSE}
Combined <- fread("~/BTSync/FetalRNAseq/Cufflinks/Combined.tracking", header=FALSE)
AllTranscripts <- ifelse(Combined[,5:ncol(Combined), with=FALSE] == '-', 0, 1)
Combined$Count<- rowSums(AllTranscripts)

```

# Cufflinks
## Transcript saturation (mean of 100 random samples) 
`r margin_note("- 207,000 match reference (=)")`
`r margin_note("- 216,000 intronic (i)")`
`r margin_note("- 77,000 fall in repeat masked regions (r)")`
`r margin_note("- 51,000 multiple classes (.)")`
`r margin_note("- 48,000 intergenic (u)")`
`r margin_note("- 36,000 splice variants (j)")`
`r margin_note("- 32,000 antisense (x)")`
`r margin_note("- 5000 pre-mRNA (e)")`
`r margin_note("- 4000 polymerse run-on (p)")`
`r margin_note("- 1000 generic exon overlap (o)")`
`r margin_note("- 89 contained (c)")`
`r margin_note("- 27 intron antisense (s)")`


```{r pressure, echo=FALSE}
#knitr::kable(
#group_by(Combined, class) %>% summarise(n())
#)
```

`r margin_note("- 'multiple classes' is the only class that shows evidence of saturation")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE, fig.height=8}
SatData <- read.delim("~/BTSync/FetalRNAseq/Cufflinks/TranscriptSat.txt", header=FALSE)
ggplot(SatData, aes(x=V2, y=V3, colour=V1)) + 
  geom_line() + 
  tufte_theme() +
  theme(legend.position=c(.9,.65)) 
 
```

`r margin_note("- all intergenic features and almost all intron antisense features are singletons")`
`r margin_note("- 'multiple classes' is bimodal, with most being confined to 2 samples or present in all")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Combined, aes(x=Count, fill=class))+geom_bar(position='dodge') +
  facet_grid(class ~ ., scales="free_y") + 
  tufte_theme() +
  scale_y_continuous(breaks=pretty_breaks(n=2), labels=comma)
```

`r margin_note("- Y-axes scaled the same. Most abundant classes excluded")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Combined, aes(x=Count, fill=class))+geom_bar(position='dodge') +
  facet_grid(class ~ .) + 
  scale_y_continuous(limits=c(0,10000), breaks=c(0,5000,10000)) +
  tufte_theme()

```

# Mixed category
`r margin_note("- intronic, repeats, intergenic and antisense are the most common types in the 'multiple types' category")`
`r margin_note("- ca. 300 reference transcripts that overlap an additional feature in another category")`
```{r include=FALSE}
Mixed <- read.delim("~/BTSync/FetalRNAseq/Cufflinks/Mixed.txt", header=FALSE)

colnames(Mixed) <- c('transfrag_id', '=', 'i', 'r', 'u', 'j', 'x', 'e', 'p', 'o', 'c', 's')
summarise_each(Mixed[,-1], funs(sum(.)))
Mixed$total <- rowSums(Mixed[,-1])
Mixed2 <- gather(Mixed, class, count, -transfrag_id, -total)

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Mixed2, aes(x=count, fill=class)) +
  geom_bar(position='dodge') +
  facet_grid(class ~ ., scales="free_y") +
  tufte_theme() +
  scale_x_continuous(limits=c(.5,25)) +
  scale_y_continuous(breaks=pretty_breaks(n=2)) +
  scale_fill_manual(values=gg_color_hue(12)[2:12])
```

`r margin_note("- Y-axes scaled the same. Most abundant classes excluded")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(Mixed2, aes(x=count, fill=class)) +
  geom_bar(position='dodge') +
  facet_grid(class ~ .) +
  scale_x_continuous(limits=c(.5,25)) +
  scale_y_continuous(limits=c(0,2000), breaks=c(0,1000,2000)) +
  tufte_theme() +
  scale_fill_manual(values=gg_color_hue(12)[2:12])


```

```{r}
my_db <- src_mysql("FetalRNAseq", host="localhost", user="root")
SampleIDlist <- collect(tbl(my_db, sql("SELECT DISTINCT sample_id FROM CufflinksGTF")))

table <- collect(tbl(my_db, sql(paste("SELECT cov, full_read_support FROM CufflinksGTF WHERE gene_id LIKE 'ENSG%' AND feature = 'transcript' AND sample_id = '", SampleIDlist$sample_id[2], "'", sep="" ))))

max <- summarise(table, max(cov)) %>% as.numeric()
cov_array <- c(0, sapply( seq(-4, floor(log10(max))), function(x) 10^x))

FP <-sapply( cov_array, function(x) {
  filter(table, cov >= x & full_read_support == 'no') %>% 
          summarise(n()) %>% 
          as.numeric()
})
TP <-sapply( cov_array, function(x) {
  filter(table, cov >= x & full_read_support == 'yes') %>% 
          summarise(n()) %>% 
          as.numeric()
})
results <- data.frame(cov=cov_array, TP = TP/TP[1], FP=FP/FP[1])
#results <- rbind(results, results2)
FP <- filter(table, full_read_support == 'no') %>% arrange(desc(cov))


ggplot(results, aes(x=FP, y=TP)) + 
  geom_line() + 
  geom_segment(x=.05, xend=.05, y=0, yend=1, colour='red', linetype='dotted') +
  annotate("text", y=.15, x=.06, hjust=0, label=paste('cov =', as.numeric(FP[floor(nrow(FP)/20),1]))) +
  tufte_theme()

Thresholds <- sapply( SampleIDlist$sample_id, function(x) {
  table <- collect(tbl(my_db, sql(paste("SELECT cov FROM CufflinksGTF WHERE gene_id LIKE 'ENSG%' AND feature = 'transcript' AND full_read_support = 'no' AND sample_id = '", x, "' ORDER BY cov DESC", sep="" ))))
  table[floor(nrow(table)/20),1]
})  
```

# Filtering on Coverage
```{r}
my_db <- src_mysql("FetalRNAseq", host="localhost", user="root")
SampleIDlist <- collect(tbl(my_db, sql("SELECT DISTINCT sample_id FROM CufflinksGTF")))

classes <- collect(tbl(my_db, sql(paste("SELECT CufflinksGTF.FPKM, CufflinksGTF.cov, CufflinksGTF.full_read_support, CufflinksGTF.end - CUfflinksGTF.start AS length, Cufflinks.class FROM CufflinksGTF, Cufflinks WHERE CufflinksGTF.transcript_id = Cufflinks.transcript_id AND feature = 'transcript' AND Cufflinks.sample_id = CufflinksGTF.sample_id AND CufflinksGTF.sample_id = '", SampleIDlist$sample_id[3], "'", sep="" ))))
```

`r margin_note("- Red line = coverage threshold")`
`r margin_note("- Coverage and FPKM are highely correlated")`
`r margin_note("- The only classes that include features without complete coverage are '=', 'c' and 'j'")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(classes, aes(x=log10(cov+0.00001), y=log10(FPKM+0.00001))) + 
  geom_point(alpha=0.1) + facet_grid(class ~full_read_support) +
  geom_segment(x=log10(0.402634), xend=log10(0.402634), y=-4, yend=4, colour='red', linetype='dotted') +
  tufte_theme() +
  xlab("log coverage") + ylab("log FPKM")
```

`r margin_note("- Red line = coverage threshold")`
`r margin_note("- VERY few features flitered out by coverage")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(classes, aes(x=log10(cov+0.00001)))+ geom_histogram() + facet_grid(class ~full_read_support, scales = 'free_y') +
  geom_segment(x=log10(0.402634), xend=log10(0.402634), y=0, yend=100000, colour='red', linetype='dotted') +
  tufte_theme() +
  xlab("log coverage)")
```

# Filtering on Length

`r margin_note("- With the exceptions of 1 macro lincRNA 1 sense intronic, 6 sense overlapping, 10 lincRNA and 11 antisense, all 76100+ lncRNAs are <10 kb")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
lncRNA_types <- c("'non_coding'", "'3prime_overlapping_ncrna'", "'antisense'", "'lincRNA'", "'bidirectional_promoter_lncrna'", "'sense_intronic'", "'sense_overlapping'", "'macro_lncRNA'")

RefFeatures <- collect(tbl(my_db, sql(paste("SELECT GencodeFeatures.`value` AS gene_type, GencodeGTF.`end` - GencodeGTF.`start` AS length FROM GencodeFeatures, GencodeGTF WHERE GencodeFeatures.`id` = GencodeGTF.`id` AND GencodeFeatures.`feature` = 'gene_type' AND GencodeGTF.`feature` = 'exon' AND GencodeFeatures.value IN (", paste0(lncRNA_types, collapse = ', ' ), ")", sep=''))))

ggplot(RefFeatures, aes(x=log10(length)))+ geom_histogram(binwidth=0.1)+facet_grid(gene_type ~ ., scales='free_y') +
  tufte_theme() +
  theme(strip.text.y = element_text(size = 8, angle = 0)) +
  geom_segment(x=log10(10000), xend=log10(10000), y=0, yend=100000, colour='red', linetype='dotted') +
  xlab("log length")
```

`r margin_note("- Intronic and intergenic are centred around 1kb, with very few exons >10kb")`
`r margin_note("- Antisense is bimodal with peaks at 1 kb and 10 kb")`
`r margin_note("- Reference lncRNA exons centred around 200 bp")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
CuffExons <- collect(tbl(my_db, sql(paste("SELECT CufflinksGTF.FPKM, CufflinksGTF.cov, CufflinksGTF.full_read_support, CufflinksGTF.end - CUfflinksGTF.start AS length, Cufflinks.class, Cufflinks.transcript_id FROM CufflinksGTF, Cufflinks WHERE CufflinksGTF.transcript_id = Cufflinks.transcript_id AND feature = 'exon' AND Cufflinks.sample_id = CufflinksGTF.sample_id AND CufflinksGTF.sample_id = '", SampleIDlist$sample_id[3], "'", sep="" ))))


ggplot(CuffExons, aes(x=log10(length)))+ geom_histogram() + 
  facet_grid(class ~ ., scales = 'free_y') +
  geom_segment(x=log10(10000), xend=log10(10000), y=0, yend=100000, colour='red', linetype='dotted') +
  tufte_theme() +
  theme(strip.text.y = element_text(size = 8, angle = 0)) +
  xlab("log length")

ggplot(CuffExons, aes(x=log10(length), y=log10(cov))) + geom_point() + facet_grid(class ~ ., scales = 'free_y')

```

# Filtering on Exon Number

`r margin_note("- Rate of single-exon transcripts is 7% for antisense (x), 10% for lincRNA (u) and 41% for sense_intronic (i)")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
RefExonNumber <- do.call(rbind, lapply( lncRNA_types, function(x) {
  table <- collect(tbl(my_db, sql(paste("SELECT GencodeFeatures.value, COUNT(GencodeFeatures.value) AS count FROM GencodeFeatures, GencodeGTF WHERE GencodeFeatures.id = GencodeGTF.id AND GencodeGTF.feature = 'exon' AND GencodeFeatures.feature = 'transcript_id' AND  GencodeFeatures.id IN ( SELECT GencodeFeatures.id FROM GencodeFeatures WHERE GencodeFeatures.feature = 'transcript_type' AND GencodeFeatures.value = ", x, " ) GROUP BY GencodeFeatures.value", sep="" ))))
  table$gene_type <- factor(x)
  table
}))

ggplot(RefExonNumber, aes(x=count)) +
  geom_histogram(binwidth=1) + 
  facet_grid(gene_type ~ ., scales = 'free_y') +
  tufte_theme() +
  theme(strip.text.y = element_text(size = 8, angle = 0)) +
  xlab("exon number")
```

`r margin_note("- Rate of single-exon transcripts WAY higher for cufflinks results.")`
`r margin_note("- There are only 108 antisense (x), 27 lincRNA (u) and 119 sense_intronic (i) novel multi-exon transcripts")`
`r margin_note("- There are also 3203 novel splice variants (j)")`
``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
CuffExonNumber <- group_by(CuffExons, transcript_id, class) %>% summarise(count = n())
ggplot(CuffExonNumber, aes(x=count)) +
  geom_histogram(binwidth=1) +
  facet_grid(class ~ ., scales = 'free_y') +
  tufte_theme() +
  theme(strip.text.y = element_text(size = 8, angle = 0)) +
  xlab("exon number")
```

`r margin_note("- Filtering out single-exon transcripts does nothing to improve the exon length distribution")`

``` {r dev.args = list(bg = 'transparent'), warning=FALSE}
CuffMultiExons <-filter(CuffExonNumber, count > 1) %>% left_join(CuffExons)
ggplot(CuffMultiExons, aes(x=log10(length)))+ geom_histogram() + 
  facet_grid(class ~ ., scales = 'free_y') +
  geom_segment(x=log10(10000), xend=log10(10000), y=0, yend=100000, colour='red', linetype='dotted') +
  tufte_theme() +
  theme(strip.text.y = element_text(size = 8, angle = 0)) +
  xlab("log length")

```
