---
title: "Data QC"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
```

## Fast QC

- All samples pass Basic stats, N content, Per-seq quality and length dist
- All samples fail or warn for per-base sequence content
- All but two fail Kmer content
    - TCTATCGGG/TGGCGCGTGC appears at start of read 1/2 40,000 + times
- Most get a warning for Overrepresented seqs but 6 fail (one pairs, four singletons )
- Only 7 samples pass GC content and only 5 pass Sequence Duplication Levels (about equal FAIL and WARN)
- Two samples from 150521 fail per tile quality (slight overloading?)
- 17921-l1_CGATGT_L006_R2_001 fails Per base quality (this is bacause the first base has low quality).
- 150610_D00125_0239_BC718AANXX_3_IL-TP-013_2.sanfastq looks horible. 150610_D00125_0239_BC718AANXX_3_IL-TP-013_1.sanfastq has fine quality, but A is really over-represented
- Edinburgh sequences have a lot of adaptor sequence

```{r summary}

summary_all <- read.delim("/Users/heo3/BTSync/FetalRNAseq/FastQC/summary.txt", header=FALSE, stringsAsFactors=FALSE)
fastQC_summary <- spread(summary_all, V2, V1)
colnames(fastQC_summary) <- c("Sample", "Adapter", "Basic_stats", "Kmers", "Overrepresented", "N_content", "Per_base_content", "Per_base_Q", "Per_seq_GC", "Per_seq_Q", "Per_tile_Q", "Duplication", "Length_dist")
summary(fastQC_summary[-1])
ggplot(summary_all, aes(x=V2, y=V3, fill=V1)) +
  geom_tile() +
  scale_fill_manual(values=c('red', 'green', 'yellow')) +
  theme(axis.text.x= element_text(angle=90, hjust = 1))

```
